### ISSUE TITLE
Create ethereum transaction to invoke a generic method of a solidity smart contract

### ISSUE THREAD

**Issue by andrea-dintino** [dyne/Zenroom#668]
My crystal ball tells me that someone will soon ask us to be able to call a generic EVM-SC method in Zenroom, therefore I propose this Zencode to be implemented (as soon as the request arrives): 

Zencode: 

```
Scenario 'ethereum': call solidity

Given I have a 'string dictionary' named 'myMethod'
Given I have a 'ethereum address' named 'solidity-address'

Given I have a  'gas price'
Given I have a 'gas limit'
Given I have an 'ethereum nonce'

Given I have an 'integer' named 'parameter1'
Given I have an 'string' named 'parameter2'

When I create the ethereum transaction to 'myMethod'

When I create the new array
When I rename the 'new array' to 'myParams'

When I insert 'parameter1' in 'myParams'
When I insert 'parameter2' in 'myParams'

# The one below is the statement to be implemented, the Zencode is temporary:
When I create the ethereum execution request for 'myMethod' using 'myParams'

Then print the 'ethereum execution request'
```

input data:

```
{
	"myMethod": {
		"method": "transfer",
		"input": [
			"address",
			"uint256"
		],
		"output": "bool"
	},
	"solidity-address": "0xE54c7b475644fBd918cfeDC57b1C9179939921E6",
	"myParams0": {
		"peppe": 1
	},
	"ethereum nonce": "0",
	"gas price": "100000000000",
	"gas limit": "300000",
	"parameter1": "123455624432432432",
	"parameter2": "pippo"
}
```

The output should look somethow like:

```
{
  "ethereum execution request": {
    "data": "b374012b0000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000003354686973206973206d7920666972737420646174612073746f726564206f6e20657468657265756d20626c6f636b636861696e00000000000000000000000000",
    "gas_limit": "300000",
    "gas_price": "100000000000",
    "nonce": "0",
    "to": "0xE54c7b475644fBd918cfeDC57b1C9179939921E6",
    "value": "0"
  }
}
```

**Comment by jaromil** [comment]
We are currently coding common function signatures of smart contracts in lua, which is easier to update when new argument sequences come up. A variadic argument interface in zencode may be too complex to implement, considering the standing solution. Good to have your opinion here @albertolerda

**Comment by albertolerda** [comment]
I don't think it is too complex, I would do something like
```
local encoder = ETH.data_contract_factory(my_method.method, my_method.input)
local transaction_data = encoder(unpack(my_params))
```

For the standard methods, we can do the parsing of the arguments only once at startup. In this case, I do it each time the zencode statement was called.

**Comment by andrea-dintino** [comment]
> Solved in #680, right?

yes, I'm closing it
